File: src/main/java/ro/mpp2024/Repository/AbstractRepository.java
package ro.mpp2024.Repository;

import ro.mpp2024.Domain.Entity;

import java.util.List;
import java.util.Optional;

public interface AbstractRepository<ID,E extends Entity<ID>>{
    public Optional<E> add(E e);
    public Optional<E> update(E e);
    public Optional<E> deleteById(ID id);
    public Optional<E> findById(ID id);
    public List<E> findAll();
}

File: SwimmingCompetitionCSharpGUI/Repository/AbstractRepository.cs
using SwimmingCompetition.Domain;

namespace SwimmingCompetition.Repository;

public interface IAbstractRepository<TId,TE> where TE: Entity<TId>
{
    public TE? Add(TE entity);
    public TE? Update(TE entity);
    public TE? DeleteById(TId id);
    public TE? FindById(TId id);
    public List<TE> FindAll();
}